node/td-utils/test/protocol-detection.test.suite.ts

```diff
@@ -0,0 +1,46 @@
+import httpAndMqtt from "./examples/httpAndMqtt.json";
+import noProtocol from "./examples/noProtocol.json";
+import onlyHttp from "./examples/onlyHttp.json";
+import onlyMqtt from "./examples/onlyMqtt.json";
+import secureProtocols from "./examples/secureProtocols.json";
+
+export const testSuite = [
+    {
+        name: "httpAndMqtt",
+        input: httpAndMqtt,
+        expected: {
+            http: [{ uri: "http://mylamp.example.com:1234" }],
+            mqtt: [{ uri: "mqtt://mylamp.example.com:1234" }],
+        },
+    },
+    {
+        name: "noProtocol",
+        input: noProtocol,
+        expected: {},
+    },
+    {
+        name: "onlyHttp",
+        input: onlyHttp,
+        expected: {
+            http: [
+                { uri: "http://mylamp.example.com:4212" },
+                { uri: "http://mylamp.example.com:4214", subprotocol: "longpoll" },
+            ],
+        },
+    },
+    {
+        name: "onlyMqtt",
+        input: onlyMqtt,
+        expected: {
+            mqtt: [{ uri: "mqtt://mylamp.example.com" }, { uri: "mqtt://mylamp.example.com", subprotocol: "longpoll" }],
+        },
+    },
+    {
+        name: "secureProtocols",
+        input: secureProtocols,
+        expected: {
+            https: [{ uri: "https://mylamp.example.com", subprotocol: "longpoll" }],
```

Comment:
@danielpeintner @relu91 should the returned binding (the first key) include the `s` or not? I can see reasons for both.

Reply:
Good point. Personally I think either we can configure it to merge http and https resulkts OR we return them separately which gives the most freedom, I think....

Reply:
We discussed in the call today and we prefer having a separate key for `http` and `https`, which only include their respective URI schemes in the objects.